### PR TITLE
feat(rds): add rds pg databases datasource

### PR DESCRIPTION
**Pull request by SabreKeyZ**
**What this PR does / why we need it**:
add rds pg databases datasource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:
**Release note**:

```release-note
add rds pg databases datasource
```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/rds/' TESTARGS='-run TestAccDatasourcePgDatabases_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccDatasourcePgDatabases_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourcePgDatabases_basic
=== PAUSE TestAccDatasourcePgDatabases_basic
=== CONT  TestAccDatasourcePgDatabases_basic
--- PASS: TestAccDatasourcePgDatabases_basic (496.26s)
PASS

```

### CODE REVIEW

File: docs/data-sources/rds_pg_databases.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `instance_id` - (Required, String) Specifies the PostgreSQL instance ID.
+
+* `database_name` - (Optional, String) Specifies the database name.
```
Comment (houpeng80) [inline docs/data-sources/rds_pg_databases.md]:
```suggestion
* `name` - (Optional, String) Specifies the database name.
```

File: huaweicloud/services/rds/data_source_huaweicloud_rds_pg_databases.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	listPgDatabasesPath = strings.ReplaceAll(listPgDatabasesPath, "{instance_id}", d.Get("instance_id").(string))
+
+	listPgDatabasesResp, err := pagination.ListAllItems(
+		listPgDatabasesClient,
+		"page",
+		listPgDatabasesPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving RDS PostgreSQL Databases")
```
Comment (houpeng80) [inline huaweicloud/services/rds/data_source_huaweicloud_rds_pg_databases.go]:
```suggestion
		return common.CheckDeletedDiag(d, err, "error retrieving RDS PostgreSQL databases")
```

File: docs/data-sources/rds_pg_databases.md (hunk 1 above)
Comment (SabreKeyZ) [reply docs/data-sources/rds_pg_databases.md]: done

File: huaweicloud/services/rds/data_source_huaweicloud_rds_pg_databases.go (hunk 2 above)
Comment (SabreKeyZ) [reply huaweicloud/services/rds/data_source_huaweicloud_rds_pg_databases.go]: done

**Pull request merged**